src/components/logger/Logger.js

```diff
@@ -0,0 +1,200 @@
+// @flow
+import * as React from 'react';
+import noop from 'lodash/noop';
+import { METRIC_TYPE_PREVIEW_METRIC, METRIC_TYPE_ELEMENTS_LOAD_METRIC } from '../../constants';
+
+type ElementsMetric = {
+    component: ElementsOrigins,
+    startMarkName?: string,
+    endMarkName: string,
+    name: string,
+    sessionId: string,
+    timestamp: string,
+    type: MetricTypes,
+};
+
+type Props = {
+    fileId?: string,
+    onMetric: (data: Object) => void,
+    children: React.ChildrenArray<React.Element<any>>,
+    source: ElementsOrigins,
+};
+
+/**
+ * Converts a character, using it as a seed, to a random integer and returns it as a string.
+ *
+ * @param {string} c - A single character
+ * @returns {string} The random character
+ */
+function generateRandom(c: string): string {
+    /* eslint-disable no-bitwise */
+    const r = (Math.random() * 16) | 0;
+    const v = c === 'x' ? r : (r & 0x3) | 0x8;
+    return v.toString(16);
+    /* eslint-enable no-bitwise */
+}
+
+/**
+ * Generates a GUID/UUID compliant with RFC4122 version 4.
+ *
+ * @return {string} A 36 character uuid
+ */
+function uuidv4() {
+    return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, generateRandom);
+}
+
+const SESSION_ID = uuidv4();
+const uniqueEvents: Set<string> = new Set();
+
+class Logger extends React.Component<Props> {
+    static defaultProps = {
+        onMetric: noop,
```

Comment:
I saw this defined in a couple other components, as well. Is it needed in both places?

Reply:
I can remove it from the components, will do that when i do the tests